homeassistant/components/nextdns/coordinator.py

```diff
@@ -0,0 +1,124 @@
+"""NextDns coordinator."""
+
+import asyncio
+from datetime import timedelta
+import logging
+from typing import TypeVar
+
+from aiohttp.client_exceptions import ClientConnectorError
+from nextdns import (
+    AnalyticsDnssec,
+    AnalyticsEncryption,
+    AnalyticsIpVersions,
+    AnalyticsProtocols,
+    AnalyticsStatus,
+    ApiError,
+    ConnectionStatus,
+    InvalidApiKeyError,
+    NextDns,
+    Settings,
+)
+from nextdns.model import NextDnsData
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+CoordinatorDataT = TypeVar("CoordinatorDataT", bound=NextDnsData)
+
+
+class NextDnsUpdateCoordinator(DataUpdateCoordinator[CoordinatorDataT]):
+    """Class to manage fetching NextDNS data API."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        nextdns: NextDns,
+        profile_id: str,
+        update_interval: timedelta,
```

Comment:
Maybe for a future PR, since the update interval differs on coordinator type, you could probably make that a class variable on the coordinator itself

Reply:
Hmm good idea, added to my todo list.